### DISCUSSION TITLE
Global override for timeouts

### DISCUSSION THREAD

**Discussion by J4Numbers** [Kong/kong#7342] [Ideas and feature requests]
### Summary

Currently, Kong enforces a 60 second timeout on all services which are connected. This timeout can be overridden by using the three variables of: `connect_timeout`, `read_timeout`, and `send_timeout` on a service level.

However, for projects which need a higher or lower timeout globally, there is no option create a default timeout which takes the place of the initial 60 seconds and every service needs to be given the override. While Kong is built on top of Nginx, which has its own timeout structures, overriding these by using `nginx_http_proxy_(connect|read|send)_timeout` has no effect on the timeout values.

As an administrator of routes, I want to set a global default timeout value, so that I can reduce configuration bloat and make it easier for other administrators who currently have to add new routes with explicit timeouts.

### Feature proposal

Add in three new options to the `kong.conf` file which are then used as defaulted overrides for the default definitions of services within the services schemas, for example:

```
proxy_default_connect_timeout = 60000
proxy_default_read_timeout    = 60000
proxy_default_send_timeout    = 60000
```

These values should be overridable at a service layer within configuration as per what currently happens.

### Impact

From what I can tell, this change would impact the following files:
* [Services schema definition](https://github.com/Kong/kong/blob/master/kong/db/schema/entities/services.lua#L38) 
* [Kong default configuration](https://github.com/Kong/kong/blob/master/kong/templates/kong_defaults.lua)
* [Configuration loader](https://github.com/Kong/kong/blob/master/kong/conf_loader/init.lua)
* [Kong example configuration](https://github.com/Kong/kong/blob/master/kong.conf.default)
* Assorted spec tests

**Comment by idbeta** [comment]
It seems to make the current settings more complicated and difficult to understand ...
